frontend/app/routes/protected/person-case/parent-details.tsx

```diff
@@ -0,0 +1,389 @@
+
+  return (
+    <>
+      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:parent-details.page-title')}</PageTitle>
+      <FetcherErrorSummary fetcherKey={fetcherKey}>
+        <fetcher.Form method="post" noValidate>
+          <input type="hidden" name="parent-amount" value={parentAmount} />
+          <ParentInformation
+            amount={parentAmount}
+            setAmount={setParentAmount}
+            defaultFormValues={defaultFormValues}
+            errors={errors}
+          />
+          <div className="mt-8 flex flex-row-reverse flex-wrap items-center justify-end gap-3">
+            <Button name="action" value="next" variant="primary" id="continue-button" disabled={isSubmitting}>
+              {t('protected:person-case.next')}
+            </Button>
+            <Button name="action" value="back" id="back-button" disabled={isSubmitting}>
+              {t('protected:person-case.previous')}
+            </Button>
+          </div>
+        </fetcher.Form>
+      </FetcherErrorSummary>
+    </>
+  );
+}
+
+type detailsType = {
+  unavailable?: boolean;
+  givenName?: string;
+  lastName?: string;
+  country?: string;
+  province?: string;
+  city?: string;
+};
+
+interface ParentInformationProps {
+  defaultFormValues?: detailsType[];
+  errors?: Record<string, [string, ...string[]] | undefined>;
+  amount: number;
+  setAmount: Dispatch<SetStateAction<number>>;
+}
+
+function ParentInformation({ defaultFormValues, errors, amount, setAmount }: ParentInformationProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const maxParents = 4;
+  const parentAmount = amount > 0 ? amount : 1;
+  const canAddParent = parentAmount < maxParents;
+
+  const onAddParent = () => {
+    if (!canAddParent) return;
+    setAmount(parentAmount + 1);
+  };
+
+  const onCloseParent = () => {
+    setAmount(parentAmount - 1);
+  };
+
+  return (
+    <>
+      <div className="space-y-10">
+        {Array.from({ length: parentAmount }).map((_, index) => (
+          <ParentForm
+            key={index}
+            index={index}
+            defaultValues={defaultFormValues?.[index]}
+            errors={errors}
+            onClose={index > 0 && index + 1 === parentAmount ? onCloseParent : undefined}
+          />
+        ))}
+      </div>
+      {canAddParent && (
+        <Button size="lg" type="button" variant="link" endIcon={faPlus} className="mt-3 px-3" onClick={onAddParent}>
+          {t('protected:parent-details.add-parent')}
+        </Button>
+      )}
+    </>
+  );
+}
+
+interface ParentFormProps {
+  index: number;
+  defaultValues?: detailsType;
+  errors?: Record<string, [string, ...string[]] | undefined>;
+  onClose?: () => void;
+}
+
+function ParentForm({ index, defaultValues, errors, onClose }: ParentFormProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [unavailable, setUnavailable] = useState<boolean>(defaultValues?.unavailable ?? false);
```

Comment:
```suggestion
  const [unavailable, setUnavailable] = useState(defaultValues?.unavailable ?? false);
```

Also, I think this variable should be named `parentDetailsUnavailable`. I had to run the application and navigate to the page to understand what `unavailable` was being used for.

Reply:
I just noticed the use of `?? false` here. This is effectively setting the default state to `false`, so why not just return `false` for this property from the loader?

Is there a reason you return `undefined` and then do a `?? false` during render, or was that just an oversight?

Sorry.. I find react code incredibly difficult to reason about sometimes.